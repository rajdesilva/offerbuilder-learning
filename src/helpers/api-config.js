import { init as initApm } from "@elastic/apm-rum";

export const baseURL = {
  API_URL_BASE: process.env.REACT_APP_BASE_API_URL,
  LOGIN_URL: process.env.REACT_APP_LOGIN_URL,
  LOGOUT_URL: process.env.REACT_APP_LOGOUT_URL,
  PROFILE_URL: process.env.REACT_APP_USER_PROFILE_URL,
  HISTORY_URL: process.env.REACT_APP_USER_HISTORY_URL,
};

export const apm = initApm({
  // Set required service name (allowed characters: a-z, A-Z, 0-9, -, _, and space)
  serviceName: process.env.REACT_APP_APM_SERVICE_NAME,
  serverUrl: process.env.REACT_APP_APM_SERVICE_URL,

  // Set service version (required for sourcemap feature)
  serviceVersion: process.env.REACT_APP_APM_SERVICE_VERSION,
});