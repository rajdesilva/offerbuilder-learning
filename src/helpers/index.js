export * from './api-config';
export * from './history';
