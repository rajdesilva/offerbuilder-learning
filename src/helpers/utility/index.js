export { createSearchQueryURL } from './createSearchQueryURL';
export { isParentOrChildrenPropertyAdded } from './isParentOrChildrenPropertyAdded';
export { isPropertySelected } from './isPropertySelected';
export { mapBrands } from './mapBrands';
export { mapOnlySuppliers } from './mapOnlySuppliers';
export { parseFormDescriptionsAndImagesForOffer } from './parseFormDescriptionsAndImagesForOffer';
export { removePropertyForSelectedChannels } from './removePropertyForSelectedChannels';
export { removePropertyForSelectedSuppliers } from './removePropertyForSelectedSuppliers';
export { getParsedProperty } from './getParsedProperty';
export { getFormattedValue } from './getFormattedValue';
export { getValue } from './getValue';
export { parsePropertyMarketingInfoToFormData } from './parsePropertyMarketingInfoToFormData';
export { createOfferRequestObject } from './createOfferRequestObject';
export { getParsedOffers } from './getParsedOffers';
export { getListOfPropertiesForSupplier } from './getListOfPropertiesForSupplier';
export { getListOfPropertiesForChannels } from './getListOfPropertiesForChannels';
export { getInitializedState } from './getInitializedState';
export { getParsedNewOfferSettingsObject } from './getParsedNewOfferSettingsObject';
export { getListOfPropertiesForStorefront } from './getListOfPropertiesForStorefront';
export { checkIfMultipleChannelPresentForTarget } from './checkIfMultipleChannelPresentForTarget';
export { checkIfMultipleSupplierPresentForTarget } from './checkIfMultipleSupplierPresentForTarget';
export { checkIfMultipleStorefrontPresentForTarget } from './checkIfMultipleStorefrontPresentForTarget';
export { removePropertyForSelectedStorefront } from './removePropertyForSelectedStorefront';
export { checkIfMultipleBrandPresentForTarget } from './checkIfMultipleBrandPresentForTarget';
export { getListOfPropertiesForBrand } from './getListOfPropertiesForBrand';
export { getSearchStateFromRedux } from './getSearchStateFromRedux';
export { getObjectStringify } from './getObjectStringify';
export { parseOfferMarketingFromOffer } from './parseOfferMarketingFromOffer';
export { parseOfferMarketingLanguagesFromOffer } from './parseOfferMarketingLanguagesFromOffer';
export { getFilterData } from './getFilterData';
export { getSearchAndFilterQueryURL } from './getSearchAndFilterQueryURL';
export { getSearchQueryURLForEditAddProperties } from "./getSearchQueryURLForEditAddProperties";
export { isSavedOfferProperty } from './isSavedOfferProperty';
export { getParsedPropertiesForDeeplinkSettings } from "./getParsedPropertiesForDeeplinkSettings";
export { resetApplicationReduxAndStorage } from "./resetApplicationReduxAndStorage";
export { updateSettingsBrandWithSearchedValues } from "./updateSettingsBrandWithSearchedValues";
export { checkIfUserHasRole } from "./checkIfUserHasRole";
export { getUserOfferBuilderRole } from "./getUserOfferBuilderRole";
export { checkIfNewPropertiesAdded } from "./checkIfNewPropertiesAdded";
export { getAllBrandsWithoutKeysFromTempBrands } from "./getAllBrandsWithoutKeysFromTempBrands";
export { checkIfItemAlreadyPresentInList } from "./checkIfItemAlreadyPresentInList";
export { checkIfTravellingDateValidAsPerBookingDate } from "./checkIfTravellingDateValidAsPerBookingDate";
export { checkIfPropertiesHaveMultipleLocations } from "./checkIfPropertiesHaveMultipleLocations";
export { displayMargin } from "./displayMargin";
export { getFilterCount } from "./getFilterCount";
export { labelForBrands } from "./labelForBrands";
export { getDistanceFromLatLonInKm } from "./getDistanceFromLatLonInKm";
export { getTrustYouIdListFromProperties } from "./getTrustYouIdListFromProperties";
export { getSearchParamsToFetchMarketPrice } from "./getSearchParamsToFetchMarketPrice";
export { getMarketPriceForPropertyFromList } from "./getMarketPriceForPropertyFromList";
export { getImageFilterCount } from "./getImageFilterCount";
export { getSearchAndFilterImagesQueryUrl } from "./getSearchAndFilterImagesQueryUrl";
export { getImageName } from "./getImageName";
export { updateReduxWithOfferData } from "./updateReduxWithOfferData";
export { getOfferTypeFromPropertiesSelected } from "./getOfferTypeFromPropertiesSelected";