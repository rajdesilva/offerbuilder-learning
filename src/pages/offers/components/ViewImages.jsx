import React, { Fragment, useState } from "react";
import { Row, Col, List, Button, Image } from "antd";
import { FormattedMessage } from "react-intl";
import defaultImg from "../../../assets/images/default-img.svg";
import styles from "./css/ViewImages.module.less";

function ViewImages(props) {
  const uploadedImages = props.isForProperty
    ? props.property.images
      ? props.property.images
      : []
    : props.images;

  const [showMore, setShowMore] = useState(false);

  return (
    <Fragment>
      <Row gutter={[8, 8]}>
        <Col span={4}>
          <FormattedMessage id="nemo.images" />
        </Col>
        <Col span={10}>
          {uploadedImages && uploadedImages.length === 0 ? (
            <Fragment />
          ) : (
            <Image.PreviewGroup>
              <List
                grid={{ gutter: 16, column: 5 }}
                dataSource={uploadedImages}
                data-testid={
                  props.isForProperty ? "prop-image-list" : "offer-image-list"
                }
                className={styles.listItemImg}
                renderItem={(image, index) => {
                  return index < 4 || uploadedImages.length === 5 ? (
                    <List.Item
                      key={index}
                      className={styles.marketingImageHover}
                    >
                      <Image
                        src={props.isForProperty ? image : image.url}
                        className={styles.listImg}
                        onError={(ev) => (ev.target.src = defaultImg)}
                      ></Image>
                    </List.Item>
                  ) : index === 4 && !showMore ? (
                    <List.Item
                      key={index}
                      className={styles.marketingImageHover}
                    >
                      <div className={styles.listItemMoreDiv}>
                        <Image
                          src={props.isForProperty ? image : image.url}
                          className={styles.blurrImage}
                          preview={false}
                          onClick={() => setShowMore(true)}
                          onError={(ev) => (ev.target.src = defaultImg)}
                        ></Image>
                        <Button
                          data-testid="show-more-btn"
                          ghost
                          type="link"
                          className={styles.showMoreImgListBtn}
                          onClick={() => setShowMore(true)}
                        >
                          {uploadedImages.length - 4}&nbsp;
                          <FormattedMessage id="nemo.more" />
                        </Button>
                      </div>
                    </List.Item>
                  ) : showMore ? (
                    <List.Item
                      key={index}
                      className={styles.marketingImageHover}
                    >
                      {index === uploadedImages.length - 1 && showMore ? (
                        <div>
                          <Image
                            src={props.isForProperty ? image : image.url}
                            className={styles.listImg}
                            onError={(ev) => (ev.target.src = defaultImg)}
                          ></Image>
                          <Button
                            ghost
                            type="link"
                            data-testid="show-less-btn"
                            className={styles.showLessBtn}
                            onClick={() => setShowMore(false)}
                          >
                            <FormattedMessage id="nemo.showLess" />
                          </Button>
                        </div>
                      ) : (
                        <Image
                          src={props.isForProperty ? image : image.url}
                          className={styles.listImg}
                          onError={(ev) => (ev.target.src = defaultImg)}
                        ></Image>
                      )}
                    </List.Item>
                  ) : (
                    <Fragment />
                  );
                }}
              />
            </Image.PreviewGroup>
          )}
        </Col>
      </Row>
    </Fragment>
  );
}

export default ViewImages;
