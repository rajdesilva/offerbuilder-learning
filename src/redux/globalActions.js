export const globalActions = {
  'SET_SUCCESS': 'SET_SUCCESS',
  'SET_ERROR': 'SET_ERROR',
  'SET_LANGUAGE': 'SET_LANGUAGE',
  'CLEAR': 'CLEAR',
}