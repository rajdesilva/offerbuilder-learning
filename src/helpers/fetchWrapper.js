import { message } from "antd";
import { createIntl, createIntlCache } from "@formatjs/intl";
import * as languages from "../langs/";
import { baseURL } from "./api-config";
import { history } from "./history";

// This is optional but highly recommended
// since it prevents memory leak
const cache = createIntlCache();

// Create the `intl` object
const intl = createIntl(
  {
    // Locale of the application
    // TODO:
    locale: "en",
    // Locale of the fallback defaultMessage
    defaultLocale: "en",
    messages: languages["en"],
  },
  cache
);
const defaultRequest = {
  credentials: "include",
  headers: {
    "Content-Type": "application/json",
  },
};
export const fetchWrapper = (url, payload) => {
  payload = {
    ...defaultRequest,
    ...payload,
  };
  return fetch(url, payload).then(
    (response) => {
      // Check that the response is valid and reject an error
      // response to prevent subsequent attempt to parse json
      if (!response.ok) {
        if (response.status === 500) {
          message.error(
            intl.formatMessage({
              id: "nemo.error500msg",
            })
          );
        } else if (response.status === 503) {
          message.error(
            intl.formatMessage({
              id: "nemo.error503msg",
            })
          );
        } else if (response.status === 504) {
          message.error(
            intl.formatMessage({
              id: "nemo.error504msg",
            })
          );
        } else {
          message.error(
            intl.formatMessage({
              id: "nemo.somethingWentWrongPleaseTryAgain",
            })
          );
        }
        return;
      }
      if (response.status === 403) {
        history.push("/access-denied");
      }
      return response.json ? response.json() : response;
    },
    (error) => {
      /**
       * user url, payload and error here for sending to the logger.
       */
      if (error instanceof TypeError) {
        window.location.href = baseURL.API_URL_BASE + "redirect";
      }
      console.error("fetch error : ", error);
    }
  );
};
