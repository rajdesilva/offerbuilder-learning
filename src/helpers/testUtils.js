import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { render as rtlRender } from "@testing-library/react";
import { getValue } from "./utility";
import rootReducers from "../redux/rootReducers";
import * as languages from "../langs";
import { IntlProvider } from "react-intl";
const initialReducerState = {};
window.getValue = getValue;

window.google = {
  maps: {
    Marker: class {},
    Map: class {},
    LatLngBounds: class {},
    places: {
      Autocomplete: class {},
      AutocompleteService: class {},
      PlacesServiceStatus: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
      PlacesAutocomplete: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
    },

    MarkerClusterer: class {},
    Geocoder: class {},
  },
};
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

function render(
  ui,
  {
    initialState = initialReducerState,
    store = createStore(rootReducers, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <IntlProvider locale="en" messages={languages["en"]}>
          {children}
        </IntlProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { render };
