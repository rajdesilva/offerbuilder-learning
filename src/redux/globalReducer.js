import { globalActions } from "./globalActions";
import produce from "immer";

const initialState = {
  error: "",
  success: "",
  language: "",
};
export const globalReducer = produce((draft, action) => {
  switch (action.type) {
    case globalActions.SET_ERROR:
      draft.success = "";
      draft.error = action.payload;
      return;
    case globalActions.SET_SUCCESS:
      draft.error = "";
      draft.success = action.payload;
      return;
    case globalActions.SET_LANGUAGE:
      draft.language = localStorage.getItem('language')
      return;
    case globalActions.CLEAR:
      return initialState;
    default:
      return draft;
  }
}, initialState);
