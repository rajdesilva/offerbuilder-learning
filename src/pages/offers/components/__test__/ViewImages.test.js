import React from "react";
import { createStore } from "redux";
import { render } from "../../../../helpers/testUtils";
import { MemoryRouter } from "react-router-dom";
import moment from "moment";
import ViewImages from "../ViewImages";

describe("View Images component test", () => {
  test("create Images components snapshot", () => {
    const tempStore = {
      searchparams: {
        latitude: "",
        longitude: "",
        startDate: "2020-10-18",
        endDate: "2020-12-17",
        los: 1,
        onlySupplier: false,
        lcn: false,
        remainingCapitalPool: null,
        pageSize: 10,
        pageOffset: 0,
        sortBy: "name",
        suppliers: [],
        brands: [
          {
            id: "idb",
            name: "Internal Demo Brand",
            storefronts: [
              {
                id: "ids",
                name: "Internal Demo Storefront",
                suppliers: [
                  {
                    id: "ntp",
                    name: "NTP",
                    channels: [
                      {
                        id: "NEMO_CH_01",
                        name: "NEMO_CH_01",
                      },
                      {
                        id: "NEMO_CH_02",
                        name: "NEMO_CH_02",
                      },
                      {
                        id: "NEMO_CH_04",
                        name: "NEMO_CH_04",
                      },
                      {
                        id: "NEMO_CH_05",
                        name: "NEMO_CH_05",
                      },
                    ],
                  },
                ],
              },
            ],
          },
        ],
        city: "",
        dateRange: {
          startDate: moment().format("YYYY-MM-DD"),
          endDate: moment().add(60, "days").format("YYYY-MM-DD"),
        },
        destination: {},
      },
      newoffersettingsparam: {
        offerId: "",
        status: {
          id: "unpublished",
          name: "Unpublished",
        },
        name: "",
        brands: [
          {
            id: "idb",
            name: "Internal Demo Brand",
            storefronts: [
              {
                id: "ids",
                name: "Internal Demo Storefront",
                suppliers: [
                  {
                    id: "ntp",
                    name: "NTP",
                    channels: [
                      {
                        id: "NEMO_CH_01",
                        name: "NEMO_CH_01",
                      },
                      {
                        id: "NEMO_CH_02",
                        name: "NEMO_CH_02",
                      },
                      {
                        id: "NEMO_CH_04",
                        name: "NEMO_CH_04",
                      },
                      {
                        id: "NEMO_CH_05",
                        name: "NEMO_CH_05",
                      },
                    ],
                  },
                ],
              },
            ],
          },
        ],
      },
      propertycart: {
        cartItems: [
          {
            img: "https://res.cloudinary.com/seekda-dev/image/upload/w_375,h_210,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/chambre_double_green_fullsize_ecodef-1140x760.jpg",
            propertyCode: "DEMO_JEEVAN",
            lcn: false,
            supplier: "ntp",
            channel: "NEMO_CH_05",
            hotelName: "AJSG Hotel",
            remainingCapital: null,
            city: "Wien",
            country: "Austria",
            rating: "",
            description:
              "Our luxurious Apartment hotel in country style is situated directly in the untouched nature of a side valley of the Vinschgau. At our house you can relax 24 hours a day: Start your day with a hearty breakfast and afterwards enjoy the beautiful nature and the various sport activities we have to offer. Relax in our alpine sauna world or our Panorama swimming pool. In the evening you can choose between three different restaurants in which we serve regional and international specialties.",
            trustyou: "",
            isParent: true,
            key: "DEMO_JEEVAN-0true",
            alternativeProperties: null,
          },
          {
            img: "https://res.cloudinary.com/seekda-dev/image/upload/w_375,h_210,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_PRATIK/PRATTT.jpg",
            propertyCode: "DEMO_PRATIK",
            lcn: false,
            supplier: "ntp",
            channel: "NEMO_CH_01",
            hotelName: "NICE RESORT PRATIK",
            remainingCapital: null,
            city: "Wien",
            country: "Austria",
            rating: 4.5,
            description:
              "Experience award-winning service and sophisticated style at JW Marriott Hotel Vienna. Ideally situated between the airport and Vienna Expressway, our 5-star hotel is an ideal destination for both business travelers and vacationing families. Luxury rooms and suites showcase deluxe bedding and marble bathrooms, as well as high-speed internet, desks with ergonomic chairs and laptop-sized safes. Many accommodations also provide striking views of the Pune cityscape or the nearby mountain. Delightful dining options within the hotel include Italian fare, a vegetarian restaurant and a chic rooftop bar. Challenge yourself to a workout in the fitness center or a swim in the outdoor pool, followed by a treatment at the tranquil Quan Spa. If you're hosting an event in Pune, you can explore our 40,000 square feWe are always looking forward to children in our hotel. In our 120 sqm large playground your children will be supervised by professional trained staff. In the meantime the parents can just chill in our relax garden or in our sauna. Besides we provide a fitness room, an indoor swimming pool with panorama view and a cosmetic studio. In summer we offer guided excursions in the beautiful mountain world starting directly in front of our house and in winter you can explore the surrounding ski area with our guides.et of versatile, sophisticated venue space, including multiple boardrooms and a ballroom that can accommodate 2,000 people. We look forward to making your visit truly remarkable at JW Marriott Hotel Vienna.",
            trustyou: "",
            isParent: true,
            key: "DEMO_PRATIK-1true",
            alternativeProperties: null,
          },
        ],
      },
      languageinfo: {
        loading: false,
        languages: [
          {
            id: "EN",
            name: "English",
          },
          {
            id: "ES",
            name: "Spanish",
          },
          {
            id: "FR",
            name: "French",
          },

          {
            id: "EL",
            name: "Greek",
          },
          {
            id: "SR",
            name: "Serbian",
          },
          {
            id: "BE",
            name: "Belarusian",
          },
          {
            id: "BG",
            name: "Bulgarian",
          },
          {
            id: "MK",
            name: "Macedonian",
          },
          {
            id: "RU",
            name: "Russian",
          },
          {
            id: "UK",
            name: "Ukrainian",
          },
          {
            id: "IW",
            name: "Hebrew",
          },
          {
            id: "AR",
            name: "Arabic",
          },
          {
            id: "HI",
            name: "Hindi",
          },
          {
            id: "TH",
            name: "Thai",
          },
          {
            id: "ZH",
            name: "Chinese",
          },
          {
            id: "JA",
            name: "Japanese",
          },
          {
            id: "KO",
            name: "Korean",
          },
        ],
      },
      newoffermarketinginfo: {
        loading: false,
        selectedLanguages: [
          {
            id: "EN",
            name: "English",
          },
        ],
      },
    };

    const store = createStore(() => ({ ...tempStore }));
    const props = {
      isForProperty: true,
      property: {
        images: [
          "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/chambre_double_green_fullsize_ecodef-1140x760.jpg",
        ],
        propertyCode: "123",
      },
      images: [],
      onChange: () => {},
    };
    const wrapper = render(
      <MemoryRouter>
        <ViewImages {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );
    const { container } = wrapper;
    expect(container.firstChild).toMatchSnapshot();
  });

  test("View Images - check if when offer images are present it renders successfully", () => {
    const tempStore = {
      searchparams: {
        latitude: "",
        longitude: "",
        startDate: "2020-10-18",
        endDate: "2020-12-17",
        los: 1,
        onlySupplier: false,
        lcn: false,
        remainingCapitalPool: null,
        pageSize: 10,
        pageOffset: 0,
        sortBy: "name",
        suppliers: [],
        brands: [
          {
            id: "idb",
            name: "Internal Demo Brand",
            storefronts: [
              {
                id: "ids",
                name: "Internal Demo Storefront",
                suppliers: [
                  {
                    id: "ntp",
                    name: "NTP",
                    channels: [
                      {
                        id: "NEMO_CH_01",
                        name: "NEMO_CH_01",
                      },
                      {
                        id: "NEMO_CH_02",
                        name: "NEMO_CH_02",
                      },
                      {
                        id: "NEMO_CH_04",
                        name: "NEMO_CH_04",
                      },
                      {
                        id: "NEMO_CH_05",
                        name: "NEMO_CH_05",
                      },
                    ],
                  },
                ],
              },
            ],
          },
        ],
        city: "",
        dateRange: {
          startDate: moment().format("YYYY-MM-DD"),
          endDate: moment().add(60, "days").format("YYYY-MM-DD"),
        },
        destination: {},
      },
      newoffersettingsparam: {
        offerId: "",
        status: {
          id: "unpublished",
          name: "Unpublished",
        },
        name: "",
        brands: [
          {
            id: "idb",
            name: "Internal Demo Brand",
            storefronts: [
              {
                id: "ids",
                name: "Internal Demo Storefront",
                suppliers: [
                  {
                    id: "ntp",
                    name: "NTP",
                    channels: [
                      {
                        id: "NEMO_CH_01",
                        name: "NEMO_CH_01",
                      },
                      {
                        id: "NEMO_CH_02",
                        name: "NEMO_CH_02",
                      },
                      {
                        id: "NEMO_CH_04",
                        name: "NEMO_CH_04",
                      },
                      {
                        id: "NEMO_CH_05",
                        name: "NEMO_CH_05",
                      },
                    ],
                  },
                ],
              },
            ],
          },
        ],
      },
      propertycart: {
        cartItems: [
          {
            img: "https://res.cloudinary.com/seekda-dev/image/upload/w_375,h_210,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/chambre_double_green_fullsize_ecodef-1140x760.jpg",
            propertyCode: "DEMO_JEEVAN",
            lcn: false,
            supplier: "ntp",
            channel: "NEMO_CH_05",
            hotelName: "AJSG Hotel",
            remainingCapital: null,
            city: "Wien",
            country: "Austria",
            rating: "",
            description:
              "Our luxurious Apartment hotel in country style is situated directly in the untouched nature of a side valley of the Vinschgau. At our house you can relax 24 hours a day: Start your day with a hearty breakfast and afterwards enjoy the beautiful nature and the various sport activities we have to offer. Relax in our alpine sauna world or our Panorama swimming pool. In the evening you can choose between three different restaurants in which we serve regional and international specialties.",
            trustyou: "",
            isParent: true,
            key: "DEMO_JEEVAN-0true",
            alternativeProperties: null,
          },
          {
            img: "https://res.cloudinary.com/seekda-dev/image/upload/w_375,h_210,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_PRATIK/PRATTT.jpg",
            propertyCode: "DEMO_PRATIK",
            lcn: false,
            supplier: "ntp",
            channel: "NEMO_CH_01",
            hotelName: "NICE RESORT PRATIK",
            remainingCapital: null,
            city: "Wien",
            country: "Austria",
            rating: 4.5,
            description:
              "Experience award-winning service and sophisticated style at JW Marriott Hotel Vienna. Ideally situated between the airport and Vienna Expressway, our 5-star hotel is an ideal destination for both business travelers and vacationing families. Luxury rooms and suites showcase deluxe bedding and marble bathrooms, as well as high-speed internet, desks with ergonomic chairs and laptop-sized safes. Many accommodations also provide striking views of the Pune cityscape or the nearby mountain. Delightful dining options within the hotel include Italian fare, a vegetarian restaurant and a chic rooftop bar. Challenge yourself to a workout in the fitness center or a swim in the outdoor pool, followed by a treatment at the tranquil Quan Spa. If you're hosting an event in Pune, you can explore our 40,000 square feWe are always looking forward to children in our hotel. In our 120 sqm large playground your children will be supervised by professional trained staff. In the meantime the parents can just chill in our relax garden or in our sauna. Besides we provide a fitness room, an indoor swimming pool with panorama view and a cosmetic studio. In summer we offer guided excursions in the beautiful mountain world starting directly in front of our house and in winter you can explore the surrounding ski area with our guides.et of versatile, sophisticated venue space, including multiple boardrooms and a ballroom that can accommodate 2,000 people. We look forward to making your visit truly remarkable at JW Marriott Hotel Vienna.",
            trustyou: "",
            isParent: true,
            key: "DEMO_PRATIK-1true",
            alternativeProperties: null,
          },
        ],
      },
      languageinfo: {
        loading: false,
        languages: [
          {
            id: "EN",
            name: "English",
          },
          {
            id: "ES",
            name: "Spanish",
          },
          {
            id: "FR",
            name: "French",
          },

          {
            id: "EL",
            name: "Greek",
          },
          {
            id: "SR",
            name: "Serbian",
          },
          {
            id: "BE",
            name: "Belarusian",
          },
          {
            id: "BG",
            name: "Bulgarian",
          },
          {
            id: "MK",
            name: "Macedonian",
          },
          {
            id: "RU",
            name: "Russian",
          },
          {
            id: "UK",
            name: "Ukrainian",
          },
          {
            id: "IW",
            name: "Hebrew",
          },
          {
            id: "AR",
            name: "Arabic",
          },
          {
            id: "HI",
            name: "Hindi",
          },
          {
            id: "TH",
            name: "Thai",
          },
          {
            id: "ZH",
            name: "Chinese",
          },
          {
            id: "JA",
            name: "Japanese",
          },
          {
            id: "KO",
            name: "Korean",
          },
        ],
      },
      newoffermarketinginfo: {
        loading: false,
        selectedLanguages: [
          {
            id: "EN",
            name: "English",
          },
        ],
      },
    };

    const store = createStore(() => ({ ...tempStore }));
    const props = {
      isForProperty: false,
      property: {
        images: [
          "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/chambre_double_green_fullsize_ecodef-1140x760.jpg",
        ],
        propertyCode: "123",
      },
      images: [
        {
          id: "15apr2021_08.jpeg",
          uploadedBy: "pradeep.misra@kognitiv.com",
          checksum: "0fb5899505e44e961de85b1630800ccb",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1618899706/offer_builder_dev/15apr2021_08.jpg",
          createdAt: "2021-04-20T08:21:47.20855",
          updatedAt: "2021-04-20T08:21:47.20855",
          sequence: 1,
        },
      ],
      onChange: () => {},
    };
    const { queryByTestId, getByTestId } = render(
      <MemoryRouter>
        <ViewImages {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );

    expect(queryByTestId("prop-image-list")).not.toBeInTheDocument();
    expect(getByTestId("offer-image-list")).toBeInTheDocument();
  });
});
