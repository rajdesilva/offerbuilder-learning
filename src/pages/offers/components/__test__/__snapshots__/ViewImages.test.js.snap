// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Images component test create Images components snapshot 1`] = `
<div
  class="ant-row"
  style="margin: -4px -4px 4px -4px;"
>
  <div
    class="ant-col ant-col-4"
    style="padding: 4px 4px 4px 4px;"
  >
    Images
  </div>
  <div
    class="ant-col ant-col-10"
    style="padding: 4px 4px 4px 4px;"
  >
    <div
      class="ant-list ant-list-split ant-list-grid"
      data-testid="prop-image-list"
    >
      <div
        class="ant-spin-nested-loading"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-row"
            style="margin-left: -8px; margin-right: -8px;"
          >
            <div
              style="width: 20%; max-width: 20%;"
            >
              <div
                class="ant-col"
                style="padding-left: 8px; padding-right: 8px; flex: 1 1 auto;"
              >
                <div
                  class="ant-list-item"
                >
                  <div
                    class="ant-image"
                  >
                    <img
                      class="ant-image-img"
                      src="https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/chambre_double_green_fullsize_ecodef-1140x760.jpg"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
